com.unity.formats.fbx/Tests/FbxTests/ModelExporterTest.cs

```diff
@@ -731,26 +739,125 @@ public class SkinnedMeshTestDataClass
             Debug.LogWarningFormat ("Compared {0} out of a possible {1} bone weights", comparisonCount, minVertCount);
         }
 
-        public class Vector3Comparer : IComparer<Vector3>
+        private delegate float GetDistance<T>(T x, T y);
+        private static float ComputeHausdorffDistance<T>(T[] orig, T[] converted, GetDistance<T> getDistance)
+        {
+            Assert.AreEqual(orig.Length, converted.Length);
+            // Compute the Hausdorff distance to determine if two meshes have the same vertices as
+            // we can't rely on the vertex order matching.
+            float maxDistance = 0;
+            for (int j = 0; j < orig.Length; j++)
+            {
+                float minDistance = float.PositiveInfinity;
+                var pos = orig[j];
+                for (int k = 0; k < orig.Length; k++)
+                {
+                    // find closest vertex in convertedMeshes
+                    var convertedPos = converted[k];
+
+                    var distance = getDistance(pos, convertedPos);
+                    if (distance < minDistance)
+                    {
+                        minDistance = distance;
+                    }
+                }
+                if (minDistance > maxDistance)
+                {
+                    maxDistance = minDistance;
+                }
+            }
+            return maxDistance;
+        }
+
+        private void TestFbxShapeNamesNotEmpty(FbxNode node)
```

Comment:
Please comment with the bug or PR number that laszlo made for reference of why we are testing this.